metabase/metabase
# Add unauthed state to dashboard parameter mapping ui

https://github.com/metabase/metaboat/issues/144

This PR adds a new unauthed state to `DashCardCardParameterMapper` when a user does not have self service data permissions.

I've also rewritten the component and replaced its usage of `PopoverWithTrigger` with a `TippyPopover`. E2E test failures make swapping popovers very annoying, so I've added some code to `Popover.jsx` so that the cypress util function `popover` can be used for both new and old popovers.


**Testing**
happy path
1. Create a dashboard that contains a Category parameter and a saved question Products table
2. Log in as a user that does not have self service data permissions for the Sample Dataset and navigate to the dashboard
3. Edit the dashboard and then click on the parameter. The card should change to a state the looks like the below screenshots.

other paths
1. as a user with self service data permissions (an admin or something), make sure you can still update the parameter's mapping
2. as a user with self service data permissions (an admin or something), add a second Location > City parameter and make sure that the Products table card shows a disabled state and has a tooltip explaining the lack of mappable fields.
3. as a user with self service data permissions (an admin or something), create a native question with a variable template tag (not a field filter) and add it to the dashboard. Map a parameter to the filter and make sure `This field only accepts a single value because it's used in a SQL query.` still shows up

<img width="605" alt="Screen Shot 2021-12-03 at 6 09 06 PM" src="https://user-images.githubusercontent.com/13057258/144693186-d344b7f2-3573-4791-b346-9fb339566ebb.png">

<img width="270" alt="Screen Shot 2021-12-03 at 6 04 28 PM" src="https://user-images.githubusercontent.com/13057258/144693425-60b458d3-6924-4fc3-9119-119778f3e6b8.png">
